forte/pymodule.py

```diff
@@ -444,10 +444,21 @@ def forte_driver(state_weights_map, scf_info, options, ints, mo_space_info):
     # solver for dynamical correlation from DSRG
     correlation_solver_type = options.get_str('CORRELATION_SOLVER')
     if correlation_solver_type != 'NONE':
+
         dsrg_proc = ProcedureDSRG(active_space_solver, state_weights_map, mo_space_info, ints, options, scf_info)
         return_en = dsrg_proc.compute_energy()
         dsrg_proc.print_summary()
         dsrg_proc.push_to_psi4_environment()
+
+        if options.get_str('DERTYPE') == 'FIRST' and active_space_solver_type == "CAS":
+            # Compute coupling coefficients
+            # NOTE: 1. Orbitals have to be semicanonicalized already to make sure
+            #          DSRG reads consistent CI coefficients before and after SemiCanonical class.
+            #       2. This is OK only when running ground-state calculations
+            state = list(state_map.keys())[0]
+            psi4.core.print_out(f"\n  ==> Coupling Coefficients for {state} <==")
+            ci_vectors = active_space_solver.eigenvectors(state)
+            dsrg_proc.compute_gradient(ci_vectors)
```

Comment:
Add `else` statement and let the user know we must have semicanonical orbitals.

Reply:
added